packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -211,6 +211,24 @@ export function buildExtraViteConfig(
 	// @ts-ignore
 	extraViteConfig.ssr = buildSSROptionsForSvelte(svelteDeps, options, config);
 
+	if (configEnv.command === 'serve') {
+		// during dev, we have to externalize transitive cjs only dependencies, see https://github.com/sveltejs/vite-plugin-svelte/issues/281
+		const optimizedTransitiveDeps = extraViteConfig.optimizeDeps?.include
+			?.filter((x) => x.includes('>'))
+			.map((x) => x.substring(x.lastIndexOf('>') + 1).trim());
+		if (optimizedTransitiveDeps?.length) {
+			// @ts-expect-error ssr still flagge in vite
+			if (!extraViteConfig.ssr.external) {
+				// @ts-expect-error ssr still flagge in vite
+				extraViteConfig.ssr.external = [];
+			}
+			// @ts-expect-error ssr still flagge in vite
+			extraViteConfig.ssr.external.push(
+				// @ts-expect-error ssr still flagge in vite
+				...optimizedTransitiveDeps.filter((x) => !config.ssr?.noExternal?.includes(x))
+			);
+		}
+	}
```

Comment:
I think we could move this into `buildSSROptionsForSvelte()` and build the list of externals from scratch, since if `prebundleSvelteLibraries` is turned off, there won't be `parent > child` syntax to handle.

Reply:
moved it and cleaned up. not sure i follow the prebundleSvelteLibraries problem.

Reply:
If `prebundleSvelteLibraries` is set to `true`, `buildOptimizeDepsForSvelte()` will early return here:

https://github.com/sveltejs/vite-plugin-svelte/blob/681002cf52cea1e0e5b3c8439151dcffe26aa5c0/packages/vite-plugin-svelte/src/utils/options.ts#L244-L252

Skipping the below code that handles adding `parent > child` to `optimizeDeps.include` so there won't be any `>` syntax to handle in configuring `ssr.external` later on, which brings up the initial issue where nested deps (e.g. `child`) weren't added to `ssr.external` when it should.

To showcase this visually with the current PR's changes:

If `prebundleSvelteLibraries: false` (default)
  - `optimizeDeps.include: ["svelte-dep > cjs-dep"]`
  - `optimizeDeps.exclude: ["svelte-dep"]`
  - `ssr.noExternal: ["svelte-dep"]`
  - `ssr.external: ["cjs-dep"]`

If `prebundleSvelteLibraries: true`
  - `optimizeDeps.include: []`
  - `optimizeDeps.exclude: []`
  - `ssr.noExternal: ["svelte-dep"]`
  - `ssr.external: []` <- should have `cjs-dep`
  
So we have to reconstruct the `ssr.external` without relying on `optimizeDeps.include` 😵